### ISSUE TITLE
Are you planning to release a Faireq Version of the same.

### ISSUE THREAD

**Issue by spprabhu** [zomux/lanmt#6]
Can you give special inputs on how to adapt it to Fairseq version.

**Comment by zomux** [comment]
Do you mean that you want to know how to train models pre-processed by fairseq?

**Comment by spprabhu** [comment]
Yes and also If I want to adapt this model to Fairseq coz its is currently using NMT and I want to move away from NMT rather use Faiseq Transformer as Base.

**Comment by spprabhu** [comment]
How will I get those KL Budgeting and other modules.Will I need to refactor the whole code.
Coz Pytorch only has LR Schedulizer.

**Comment by zomux** [comment]
The nmtlab package is closer to openmt, so it will be pretty easy changing the nmtlab part to opennmt-py

**Comment by zomux** [comment]
KL budget is super easy to implement. It can be computed based the iteration number

**Comment by zomux** [comment]
I will investigate how fairseq preprocess texts

**Comment by spprabhu** [comment]
Sure coz Fairseq is not using the sentencepiece BPE.

**Comment by spprabhu** [comment]
Also Do you have .sp datasets for other language pairs.Have you tried running your model on the same.What were their results apart from ASPEC Dataset.For eg : En-It, En-Fr etc.

**Comment by spprabhu** [comment]
It would be helpful if you already have those datasets .

**Comment by zomux** [comment]
No I don't have them, I just took the pre-processed dataset from opennmt.

**Comment by spprabhu** [comment]
Hi Raphael do you have EN-IT dataset in the .sp format

**Comment by zomux** [comment]
My WMT14 dataset is actually obtained from openNMT, the link is here https://s3.amazonaws.com/opennmt-trainingdata/wmt14-ende-sp.tar.gz . I don't think they have a EN-IT dataset. But I guess it won't be difficult to process it using sentencepiece ---> https://github.com/google/sentencepiece

**Comment by spprabhu** [comment]
Thanks for all the help.Just one question Did you look at the previous version of this model which is already present on pytorch page.Its released in Feb 2018.
Will that help me in adapting this Updated model to Fairseq.

**Comment by zomux** [comment]
Which one, Jason's model? Can you give me a link.

**Comment by zomux** [comment]
Actually if fairseq people want, I can try to build my model into their package lol. Although I'm not familiar with fairseq API.

**Comment by spprabhu** [comment]
Yes ,The Jasons Model.https://github.com/nyu-dl/dl4mt-nonauto

**Comment by zomux** [comment]
Oh, it turns out he is not using fairseq. Anyway, I suggest you to directly use my code. ".sp" files are just subword level corpus which you can tokenize with sentencepiece. Of course you can use any tokenizer as you want. I may give you a script for extracting the vocabulary from tokenized texts.

**Comment by zomux** [comment]
And if you want to re implement with fairseq, I don't think it will be difficult. Fairseq is also based on PyTorch.

**Comment by spprabhu** [comment]
Thanks The script would be really helpful so that I can reproduce those results with same vocabulary in a different setting.

**Comment by zomux** [comment]
The vocabulary files are already there in the files you download. “*.vocab” files are just picked Python lists. You can obtain the list by

import pickle
print(pickle.load(open(“xxx.vocab”, “rb”)))

**Comment by zomux** [comment]
I will upload the script for extracting the vocabularies.
